packages/rocketchat-ldap/server/sync.js

```diff
@@ -129,11 +129,12 @@ export function getDataToSyncUserData(ldapUser, user) {
 						// TODO: Find a better solution.
 						const dKeys = userField.split('.');
 						const lastKey = _.last(dKeys);
-						_.reduce(dKeys, (obj, currKey) =>
-							((currKey === lastKey)
+						_.reduce(dKeys, (obj, currKey) => (
+							(currKey === lastKey)
 								? obj[currKey] = tmpLdapField
-								: obj[currKey] = obj[currKey] || {})
-							, userData);
+								: obj[currKey] = obj[currKey] || {}
+							, userData)
+						);
```

Comment:
This looks strange, the userData would be the second parameter of the reduce right?

Reply:
yes.. fixing